woocommerce-gateway-stripe.php

```diff
@@ -371,17 +371,19 @@ public function plugin_row_meta( $links, $file ) {
 			}
 
 			/**
-			 * Add the gateways to WooCommerce.
+			 * Add the main Stripe gateway to WooCommerce.
 			 *
 			 * @since 1.0.0
 			 * @version 5.6.0
 			 */
 			public function add_gateways( $methods ) {
 				$methods[] = $this->get_main_stripe_gateway();
 
-				// These payment gateways will always be visible, regardless if UPE is enabled or disabled:
-				$methods[] = WC_Gateway_Stripe_Alipay::class;
-				$methods[] = WC_Gateway_Stripe_Multibanco::class;
+				// These payment gateways will be visible in the main settings page, if UPE enabled.
+				if ( WC_Stripe_Feature_Flags::is_upe_checkout_enabled() ) {
+					$methods[] = WC_Gateway_Stripe_Alipay::class;
+					$methods[] = WC_Gateway_Stripe_Multibanco::class;
+				}
```

Comment:
In this PR, Alipay and Multibanco are moved inside the Stripe settings page only for UPE disabled state.

Currently in `develop` when UPE is enabled, these 2 methods are not part of UPE. And they are rendered individually on the checkout page. 

<img src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/0b2ab399-13cb-4747-9dd3-d1aeaf7909ab" width="50%" height="50%" />


I will handle them in a separate PR for UPE enabled state.

Reply:
I'm not sure if this is related to this but if you have UPE enabled, Alipay and Multibanco settings aren't shown anywhere - either in the Payments table or the Stripe Payment methods list.

| <img width="585" alt="Screenshot 2024-01-31 at 1 39 22 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/3384b170-1d99-45cf-8141-a39e384dc7f6"> | <img width="693" alt="Screenshot 2024-01-31 at 1 39 00 pm" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/71a6c7c4-b4d5-43cc-87c4-1c27793c44e5"> |
|--------|--------|

Is this the part you mean you'll address separately?

Reply:
> Is this the part you mean you'll address separately?

Not exactly this. In a separate PR I will move them inside the Stripe settings page but for now you should see them on the main payment settings page (For UPE enabled). I see them on the payment settings page -

![upe enabled](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/92f48b8e-d9ee-44d5-9945-86c3a49090fa)

Could you please check if you have these changes in your local? :eyes:  
https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2853/commits/fa6f6ab382f3c7a5075d4062bcd86278bcb5802a

Reply:
Yep that was it. Thanks @Mayisha.